projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -185,13 +185,76 @@ public function build( $id ) {
 
 			// if task can't be found don't add anything
 			if ( ! empty( $task ) ) {
-				$tasks_for_task_list[] = $task;
+				$tasks_for_task_list[] = $this->build_task( $task );
 			}
 		}
 
 		return $tasks_for_task_list;
 	}
 
+	/**
+	 * Builds a single task with current state
+	 *
+	 * @param Task $task Task definition.
+	 * @return Task Task with current state.
+	 */
+	private function build_task( $task ) {
+		$built_task                 = array(
+			'id'    => $task['id'],
+			'title' => $task['title'],
+		);
+		$built_task['completed']    = $this->is_task_complete( $task );
+		$built_task['disabled']     = $this->is_task_disabled( $task );
+		$built_task['subtitle']     = $this->load_subtitle( $task );
+		$built_task['isLaunchTask'] = isset( $task['isLaunchTask'] ) ? $task['isLaunchTask'] : false;
+		$built_task['badge_text']   = isset( $task['badge_text'] ) ? $task['badge_text'] : '';
+		return $built_task;
+	}
+
+	/**
+	 * Loads a subtitle for a task, calling the callback if it exists.
+	 *
+	 * @param Task $task A task definition.
+	 * @return string The subtitle for the task.
+	 */
+	private function load_subtitle( $task ) {
+		if ( isset( $task['subtitle'] ) ) {
+			if ( is_callable( $task['subtitle'] ) ) {
+				return call_user_func( $task['subtitle'] );
+			}
+			return $task['subtitle'];
+		}
+		return '';
+	}
+
+	/**
+	 * Checks if a task is disabled
+	 *
+	 * @param array $task Task definition.
+	 * @return boolean
+	 */
+	public function is_task_disabled( $task ) {
+		if ( isset( $task['is_disabled'] ) && is_callable( $task['is_disabled'] ) ) {
+			return call_user_func( $task['is_disabled'] );
+		}
+		return false;
```

Comment:
`NB:`

Should we return what is defined on the task, if it's a boolean, instead of defaulting to false?

`return is_bool( $task['is_disabled'] ) ? $task['is_disabled'] : false`

Reply:
The idea here was that `is_disabled` would _only_ be a callback to prevent trying to evaluate things at definition time, which is almost always too early.

Reply:
In that case, should we rename `is_disabled` to `is_disabled_callback`?

Reply:
That is a solid idea and this is just the right time to make a change like that